R/sim_gs_n.R

```diff
@@ -242,15 +233,29 @@ sim_gs_n <- function(
     block = rep(c("experimental", "control"), 2),
     test = wlr,
     cut = NULL,
-    seed = 2024,
     ...) {
   # Input checking
   # TODO
 
+  # parallel computation message for backends ----
+  if (!is(plan(), "sequential")) {
+    # future backend
+    message("Using ", nbrOfWorkers(), " cores with backend ", attr(plan("list")[[1]], "class")[2])
+  } else if (foreach::getDoParWorkers() > 1) {
+    message("Using ", foreach::getDoParWorkers(), " cores with backend ", foreach::getDoParName())
+    message("Warning: ")
+    message("doFuture may exhibit suboptimal performance when using a doParallel backend.")
+  } else {
+    message("Backend uses sequential processing.")
+  }
+
   # Simulate for `n_sim` times
-  ans <- NULL
-  for (sim_id in seq_len(n_sim)) {
-    set.seed(seed + sim_id)
+  ans <- foreach::foreach(
+    sim_id = seq_len(n_sim),
+    .combine = "rbind",
+    .errorhandling = "stop",
```

Comment:
In `sim_fixed()` you have `.errorhandling = "pass"`. Is using `"stop"` here intentional?

Reply:
This difference changes the behavior of expected errors. Using the `"pass"` option returns the error instead of the anticipated result whereas `"stop"` will return only the error. If we use the `"pass"` option, it will change the intended behavior of the `stop("If you want...")` error that sim_gs_n throws to return a data frame of these errors, rather than a single error as it currently functions

Reply:
OK! Good to know. I think it may be useful to document this behaviour clearly, as this is quite subtle and differs from the behavior in `sim_fixed_n()`. Do you mind adding something about this to the roxygen section?

Reply:
Added comments to the roxygen section in `sim_gs_n`. I have updated the `sim_fixed_n()` to use `"stop"` for parity.

Reply:
Nice